### ISSUE TITLE
Loss based training.

### ISSUE THREAD

**Issue by Canxes** [fchollet/keras#8663]
Hello!
During my experiments with CNN I found out that it's rather useful to pick samples for the next batch based upon they loss history. In other words, the idea is to model some non uniform distribution on train samples, where probability for each sample is function of it's loss history. 

I ended up with some custom and task-specific solution, but I wonder is it interesting for the keras community? If so, should it be a feature for keras or keras-contrib?

I have some ideas about API, but first I need to understand is it really good to have this feature in keras. 

P.S. Also I would appreciate any references to papers that mention such training approach.

**Comment by fchollet** [comment]
Adaptative sampling during training is broadly useful. If you can come up with a design that generalizes to multiple approaches, I suggest that you submit it to Keras-contrib, and we can move it to core Keras a little bit later, after it matures.

**Comment by fchollet** [comment]
> P.S. Also I would appreciate any references to papers that mention such training approach.

There are many, but here's one I liked at the time (2015): https://arxiv.org/abs/1506.09016
